src/binding.cc

```diff
@@ -1013,6 +1015,33 @@ class ExternalStaticOneByteStringResource
   const int _length;
 };
 
+// NOTE: This class is never used and only serves as a reference for
+// the OneByteConst struct created on Rust-side.
+class ExternalConstOneByteStringResource
+    : public v8::String::ExternalOneByteStringResource {
+ public:
+  ExternalConstOneByteStringResource(int length)
+      : _length(length) {
+    static_assert(offsetof(ExternalConstOneByteStringResource, _length) == 16,
+                  "ExternalConstOneByteStringResource's length was not at offset 16");
+    static_assert(sizeof(ExternalConstOneByteStringResource) == 24,
+                  "ExternalConstOneByteStringResource size was not 24");
+    static_assert(alignof(ExternalConstOneByteStringResource) == 8,
+                  "ExternalConstOneByteStringResource align was not 8");
```

Comment:
FWIW, the 32-bit build started failing here:

```
   ../../../src/binding.cc:68:1: error: static assertion failed due to requirement 'sizeof(v8::Isolate::DisallowJavascriptExecutionScope) == sizeof(unsigned int) * 2': DisallowJavascriptExecutionScope size mismatch
   static_assert(sizeof(v8::Isolate::DisallowJavascriptExecutionScope) ==
   ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ../../../src/binding.cc:68:69: note: expression evaluates to '12 == 8'
   static_assert(sizeof(v8::Isolate::DisallowJavascriptExecutionScope) ==
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
   ../../../src/binding.cc:1033:5: error: static assertion failed due to requirement '__builtin_offsetof(ExternalConstOneByteStringResource, _length) == 16': ExternalConstOneByteStringResource's length was not at offset 16
       static_assert(offsetof(ExternalConstOneByteStringResource, _length) == 16,
       ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ../../../src/binding.cc:1033:73: note: expression evaluates to '8 == 16'
       static_assert(offsetof(ExternalConstOneByteStringResource, _length) == 16,
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
   ../../../src/binding.cc:1035:5: error: static assertion failed due to requirement 'sizeof(ExternalConstOneByteStringResource) == 24': ExternalConstOneByteStringResource size was not 24
       static_assert(sizeof(ExternalConstOneByteStringResource) == 24,
       ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ../../../src/binding.cc:1035:62: note: expression evaluates to '12 == 24'
       static_assert(sizeof(ExternalConstOneByteStringResource) == 24,
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
   ../../../src/binding.cc:1037:5: error: static assertion failed due to requirement 'alignof(ExternalConstOneByteStringResource) == 8': ExternalConstOneByteStringResource align was not 8
       static_assert(alignof(ExternalConstOneByteStringResource) == 8,
       ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ../../../src/binding.cc:1037:63: note: expression evaluates to '4 == 8'
       static_assert(alignof(ExternalConstOneByteStringResource) == 8,
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
```

Reply:
@avindra Thanks for letting us know and sorry about the break! Can you check if this PR fixes the build on 32-bit architectures? https://github.com/denoland/rusty_v8/pull/1289

Reply:
> Can you check if this PR fixes the build on 32-bit architectures? #1289

Sorry for the late reply. The 32 bit build passes now!